### DISCUSSION TITLE
Customizable "send" URL

### DISCUSSION THREAD

**Discussion by ajgon** [dani-garcia/vaultwarden#4361] [Ideas]
So here is my use-case - I have my whole vaultwarden instance hidden inside my private network. However it would be great, to expose send links (and only them) to the Internet in case I wish to share something with somebody who is not connected to my VPN. Currently send links are just root (`/`) slashes with some hashtag-javascript-magic, which will require me to expose whole vaultwarden basically.

Would it be possible, to have a configurable "send" path? Or (if it's easier) configurable "send" domain? Currently I have separate domain for "send" and it works like a charm, but I have to manually copy the link every time, replace the main vaultwarden domain with "send" domain, and then send a link which is cumbersome.

Thank you, for this amazing project!

**Comment by stefan0xC** [comment]
I'm not sure this is possible. First of all, this would have to be done client-side as [the generation and handling of Send links is done entirely client-side](https://bitwarden.com/blog/bitwarden-send-how-it-works/#understanding-a-send-link). So this is not something we control (as we only minimally patch the web-vault to make it compatible with Vaultwarden). 

And as far as I've looked into it, this is not configurable for the web-vault either. Bitwarden also just [relies on "hashtag-javascript-magic"](https://send.bitwarden.com/script.js) to redirect send.bitwarden.com links to the proper web-vault (but this approach wouldn't help you if you don't want to expose the web-vault).

**Comment by ajgon** [comment]
So maybe only configuratble "send" domain? i.e. `SEND_DOMAIN=send.example.com` and the only place which it needs be used, is where the "send" links are shown/shared to the user? Unless they're also hardcoded in client? Thanks to that, everything else will remain the same.

**Reply by stefan0xC** [reply]
If you just don't want to manually copy the link, you can look at the bitwarden/clients repository and build the web-vault with [the necessary changes](https://github.com/bitwarden/clients/blob/cbcd5d9747d4f1f5f54be8a2ce82e0a356343c58/apps/cli/src/tools/send/commands/receive.command.ts#L113-L114) as they have done yourself (i.e. that the clients will generate the correct URL)

**Reply by ajgon** [reply]
But this is the API link, this one is fine (unless I miss something). The thing is, I have two domains pointing to the same vaultwarden instance, let's say `vault.example.com` and `send.example.com`. Both work perfectly on their own. The only thing I need, to make this setup to work is `vault.example.com` generating "share" links for send as `send.example.com`.

**Reply by ajgon** [reply]
Nevermind, it occured to me, that this wouldn't make sense, as send.example.com will be also fully exposed, just under different domain 😅

**Comment by RealOrangeOne** [comment]
It's worth noting that what you see in the browser URL doesn't necessarily correspond to what you'd want to expose. The web vault and API are separate components, and really it's the API you want to isolate. Simply isolating `/send` isn't going to work as the API isn't accessible at `/send`.

**Reply by ajgon** [reply]
Yes, that makes total sense, so we can rule out the "path" option. But what about domain? So `vault.example.com` generating "share/send" links as `https://send.example.com/#/send/.....`.

**Reply by ajgon** [reply]
Okay, but now I'm exposing whole API under send which is missing the point 😅 . Okay, I think it is what it is then, thanks for clearing this up!

**Reply by RealOrangeOne** [reply]
> https://send.example.com/#/send/.....

That little `#` in there means it's not _actually_ a path, and it's actually likely your reverse proxy won't even see it.

**Reply by stefan0xC** [reply]
Shouldn't it work if you only `proxy_pass` the things the web-vault needs to display Sends and deny everything else? (I mean, the login prompt would be shown but you won't be able to login.)
